src/app-layer-ssh.h

```diff
@@ -48,40 +48,12 @@
     the lenghts and msg_code (inside payload, if any)
 */
 
-typedef struct SshHeader_ {
-    uint32_t pkt_len;
-    uint8_t padding_len;
-    uint8_t msg_code;
-    uint16_t banner_len;
-    uint8_t buf[6];
-    uint8_t buf_offset;
-    uint8_t flags;
-    uint32_t record_left;
-    uint8_t *proto_version;
-    uint8_t *software_version;
-    uint8_t *banner_buffer;
-} SshHeader;
-
 enum {
     SSH_STATE_IN_PROGRESS,
```

Comment:
Not sure about these : can we share an enumeration between Rust and C ? (cf SSHConnectionState)

Reply:
Not sure either. Do you still need it?

Reply:
`SSH_STATE_BANNER_DONE` is being used in detect-ssh-*.c, output-json-ssh.c, and output-lua.c

Maybe it is being used abusively as a shortcut to `rs_ssh_state_progress_completion_status()`

Reply:
Ah I see. So in these places it is used to make sure logging/detection happens as soon as it can, which means before the tx is considered complete, but after we have the banner parsed. So I agree we need a way to express this in the Rust parser case as well.

Reply:
Ok, and what is the Rust idiomatic way ?

Reply:
@jasonish do you know?

Reply:
Working with cbindgen.toml, I did not manage to make cbindgen produce an enum, but I can export constants.
I did not manage to make it work with the example of https://blog.eqrion.net/announcing-cbindgen/

Reply:
> Working with cbindgen.toml, I did not manage to make cbindgen produce an enum, but I can export constants.
> I did not manage to make it work with the example of https://blog.eqrion.net/announcing-cbindgen/

Apparently if the enum is not used in the signature of an export function, the enum will not be exported either. To get around this, it looks like we can make this change to cbindgen.toml:
```
diff --git a/rust/cbindgen.toml b/rust/cbindgen.toml
index f6117c403..c69469443 100644
--- a/rust/cbindgen.toml
+++ b/rust/cbindgen.toml
@@ -71,7 +71,10 @@ documentation_style = "doxy"
 # found but otherwise don't appear to be used by the public API.
 #
 # default: []
-include = ["AppLayerGetTxIterTuple"]
+include = [
+    "AppLayerGetTxIterTuple",
+    "SSHEvent",
+]
 
 # A list of items to not include in the generated bindings
 # default: []
@@ -109,7 +112,7 @@ exclude = [
 # * "functions":
 #
 # default: []
-item_types = ["structs","opaque","functions"]
+item_types = ["structs","opaque","functions","enums"]
 
 # Whether applying rules in export.rename prevents export.prefix from applying.
 #
```